rocketmad/models.py

```diff
@@ -26,883 +24,945 @@
+                ScannedLocation.latitude <= neLat,
+                ScannedLocation.longitude <= neLng
+            )
         elif oSwLat and oSwLng and oNeLat and oNeLng:
-            # Send weather in view but exclude those within old boundaries.
-            # Only send newly uncovered weather.
-            query = (query
-                     .where(
-                         ((Weather.latitude >= float(swLat) - lat_delta) &
-                          (Weather.longitude >= float(swLng) - lng_delta) &
-                          (Weather.latitude <= float(neLat) + lat_delta) &
-                          (Weather.longitude <= float(neLng) + lng_delta)) & ~
-                         ((Weather.latitude >= float(oSwLat) - lat_delta) &
-                          (Weather.longitude >= float(oSwLng) - lng_delta) &
-                          (Weather.latitude <= float(oNeLat) + lat_delta) &
-                          (Weather.longitude <= float(oNeLng) + lng_delta))))
+            # Send scanned locations in view but exclude those within old
+            # boundaries. Only send newly uncovered scanned locations.
+            query = query.filter(
+                ScannedLocation.last_modified >= active_time,
+                ScannedLocation.latitude >= swLat,
+                ScannedLocation.longitude >= swLng,
+                ScannedLocation.latitude <= neLat,
+                ScannedLocation.longitude <= neLng,
+                ~and_(
+                    ScannedLocation.latitude >= oSwLat,
+                    ScannedLocation.longitude >= oSwLng,
+                    ScannedLocation.latitude <= oNeLat,
+                    ScannedLocation.longitude <= oNeLng
+                )
+            )
+        else:
+            query = query.filter(
+                ScannedLocation.last_modified >= active_time,
+                ScannedLocation.latitude >= swLat,
+                ScannedLocation.longitude >= swLng,
+                ScannedLocation.latitude <= neLat,
+                ScannedLocation.longitude <= neLng
+            )
+
+        result = query.all()
 
+        return [loc._asdict() for loc in result]
+
+
+class RmVersion(db.Model):
+    __tablename__ = 'rmversion'
+
+    key = db.Column(
+        db.String(length=16, collation='utf8mb4_unicode_ci'), primary_key=True
+    )
+    val = db.Column(db.SmallInteger)
+
+
+def orm_to_dict(orm_result):
+    if isinstance(orm_result, list):
+        result = []
+        for r in orm_result:
+            d = dict(r.__dict__)
+            del d['_sa_instance_state']
+            result.append(d)
+    else:
+        result = dict(orm_result.__dict__)
+        del result['_sa_instance_state']
+    return result
+
+
+def table_exists(table_model):
+    return db.engine.has_table(table_model.__tablename__)
+
+
+def create_table(table_model):
+    if not table_exists(table_model):
+        table_model.__table__.create(db.engine)
+
+
+def create_rm_tables():
+    tables = []
+    for table in tables:
```

Comment:
Where the tables are coming from?

Reply:
From `tables`, but we don't have any RM tables yet (except for rmversion, but that one is created somewhere else.) :)